janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -212,6 +214,111 @@ private static void assertStorageExists(Backend backend, boolean exists) throws
         }
     }
 
+    /**
+     * Test that has(key) query utilizes mixed index if possible
+     * All data types supported by mixed index are tested
+     */
+    @Test
+    public void testHasKeyQuery() {
+        VertexLabel person = mgmt.makeVertexLabel("person").make();
+        PropertyKey name = mgmt.makePropertyKey("name").dataType(String.class).make();
+        PropertyKey nickname = mgmt.makePropertyKey("nickname").dataType(String.class).make();
+        PropertyKey age = mgmt.makePropertyKey("age").dataType(Integer.class).make();
+        PropertyKey houses = mgmt.makePropertyKey("houses").dataType(Integer.class).make();
```

Comment:
I find it easier to understand what these different properties are used for, if each one is named by the data type it uses. In addition, the selection of properties can be trimmed down to only contain _a single_ property per data type.